apps/molgenis-components/src/components/forms/formUtils/formUtils.ts

```diff
@@ -227,6 +236,21 @@ function containsInvalidPeriod(periods: any) {
   return periods.find((period: any) => !isValidPeriod(period));
 }
 
+export function parseJson(json: string) {
+  try {
+    return JSON.parse(json);
+  } catch (error) {
+    return false;
+  }
+}
+
+export function isJsonObjectOrArray(parsedJson: any) {
+  if (typeof parsedJson === "object") {
```

Comment:
doesn't check if it is an array?

Reply:
It should allow arrays:
<img width="815" alt="Scherm­afbeelding 2024-11-26 om 13 56 42" src="https://github.com/user-attachments/assets/8cccc27e-e492-4730-aeab-0f7f96f195f8">

While `null` is also an `object`, it does fail in the form (probably due to `if(!parsed)`):
<img width="1141" alt="Scherm­afbeelding 2024-11-26 om 13 58 50" src="https://github.com/user-attachments/assets/68af3ec9-b780-49e5-ad92-e33c4ce63f9d">

Reply:
I'll have a quick look if I can fix `null` giving the correct error message though (no array/object instead of invalid JSON).

Reply:
Test code for front-end now also added: https://github.com/molgenis/molgenis-emx2/blob/ff6cdbfac2bf41c497ba66328c93f3f56e1eb33a/apps/molgenis-components/src/components/forms/formUtils/formUtils.test.ts#L380-L391